File: serene/src/Serene.Web/Modules/ServerTypes/Workflow/WorkflowDefinitionRow.ts
import { fieldsProxy } from "@serenity-is/corelib";

export interface WorkflowDefinitionRow {
    Id?: number;
    WorkflowKey?: string;
    Name?: string;
    InitialState?: string;
}

export abstract class WorkflowDefinitionRow {
    static readonly idProperty = 'Id';
    static readonly nameProperty = 'WorkflowKey';
    static readonly localTextPrefix = 'Workflow.WorkflowDefinition';
    static readonly deletePermission = 'Workflow:Modify';
    static readonly insertPermission = 'Workflow:Modify';
    static readonly readPermission = 'Workflow:View';
    static readonly updatePermission = 'Workflow:Modify';

    static readonly Fields = fieldsProxy<WorkflowDefinitionRow>();
}

File: serene/src/Serene.Web/Workflow/DbProvider/Entities/WorkflowDefinitionRow.cs
using Serenity.ComponentModel;
using Serenity.Data;
using Serenity.Data.Mapping;

namespace Serene.Web.Workflow.DbProvider.Entities
{
    [ConnectionKey("Default"), Module("Workflow"), TableName("WorkflowDefinitions")]
    [DisplayName("Workflow Definitions"), InstanceName("Workflow Definition")]
    [ReadPermission("Workflow:View"), ModifyPermission("Workflow:Modify")]
    public class WorkflowDefinitionRow : Row<WorkflowDefinitionRow.RowFields>
    {
        [Identity, IdProperty]
        public int? Id { get => fields.Id[this]; set => fields.Id[this] = value; }
        public string? WorkflowKey { get => fields.WorkflowKey[this]; set => fields.WorkflowKey[this] = value; }
        [NameProperty]
        public string? Name { get => fields.Name[this]; set => fields.Name[this] = value; }
        public string? InitialState { get => fields.InitialState[this]; set => fields.InitialState[this] = value; }


        public class RowFields : RowFieldsBase
        {
            public Int32Field Id;
            public StringField WorkflowKey;
            public StringField Name;
            public StringField InitialState;
        }
    }
}
